File: 27-RemoveElement/main.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
  if len(nums) == 0 {
	return 0
  }

  i := 0
  for j := 0; j < len(nums); j++ {
	if nums[j] != val {
	  nums[i] = nums[j]
	  i++
	}
  }

  return i
}

func main() {
  fmt.Println(removeElement([]int{3, 2, 2, 3}, 3))
  fmt.Println(removeElement([]int{0, 1, 2, 2, 3, 0, 4, 2}, 2))
}

File: 27-RemoveElement/main.ts
const removeElement = (nums: number[], val: number): number => {
  if (nums.length === 0) return 0;

  let i: number = 0;
  for (let j: number = 0; j < nums.length; j++) {
    if (nums[j] !== val) {
      nums[i] = nums[j];
      i++;
    }
  }

  return i;
}

console.log(removeElement([3, 2, 2, 3], 3));
console.log(removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))